src/routes/Home/Home.tsx

```diff
@@ -1,10 +1,35 @@
+import { components } from '@/api/schema';
 import { Title } from '@/components';
 import { HeroCard } from '@jod/design-system';
 import { useTranslation } from 'react-i18next';
-import { Link } from 'react-router-dom';
+import { Link, useRouteLoaderData } from 'react-router-dom';
 
 const Home = () => {
   const { t, i18n } = useTranslation();
+  const data = useRouteLoaderData('root') as components['schemas']['YksiloCsrfDto'] | null;
+
+  let preferencesTo = `${t('slugs.profile.index')}/${t('slugs.profile.preferences')}`;
```

Comment:
Alternative way could be like the following, if wanting to avoid `let`s, I find this little more readable but either way is fine.

```tsx
  const getPreferencesLink = () => {
    const params = new URLSearchParams();
    params.set('lang', i18n.language);
    params.set('callback', `/${i18n.language}/${t('slugs.profile.index')}/${t('slugs.profile.preferences')}`);

    return data
      ? {
        to: `${t('slugs.profile.index')}/${t('slugs.profile.preferences')}`,
        component: Link,
      }
      : {
        to: `/login?${params.toString()}`,
        component: ({
          to,
          className,
          children,
        }: {
          to: object | string;
          className?: string;
          children: React.ReactNode;
        }) => (
          <a href={to as string} className={className}>
            {children}
          </a>
        ),
      };
  };
// ...
            <HeroCard
              to={getPreferencesLink().to}
              linkComponent={getPreferencesLink().component}
              size="sm"
              textColor="#000"
              backgroundColor="#CD4EB3F2"
              title={t('home.card-4-title')}
            />

// ...

```